web/libs/editor/src/tags/object/Paragraphs/HtxParagraphs.jsx

```diff
@@ -518,19 +613,216 @@ class HtxParagraphsView extends Component {
+        item.annotation.unselectArea(region);
+      }
+    });
+  };
+
+  // Helper to select all regions for a phrase index
+  selectRegionsForPhrase = (phraseIdx) => {
+    if (!isFF(FF_NER_SELECT_ALL)) return;
+
+    const item = this.props.item;
+    if (!item || !item.annotation || !item.annotation.results) return;
+
+    // Only deselect regions, not labels
+    this.unselectAllRegions();
+
+    // Get all regions for this phrase
+    const phraseRegions = this.getRegionsForPhrase(phraseIdx);
+
+    // If there are regions, automatically select the first one
+    if (phraseRegions.length > 0) {
+      this.selectRegion(phraseRegions[0]);
+    }
+  };
+
+  // Move to the next phrase
+  handleNextPhrase = () => {
+    const item = this.props.item;
+    if (!item || !item._value || item._value.length === 0) return;
+    const currentIdx = typeof item.playingId === "number" && item.playingId >= 0 ? item.playingId : -1;
+    const nextIdx = Math.min(currentIdx + 1, item._value.length - 1);
+    if (nextIdx !== currentIdx) {
+      if (item.seekToPhrase) item.seekToPhrase(nextIdx);
+      this.selectRegionsForPhrase(nextIdx);
+    }
+  };
+
+  // Move to the previous phrase
+  handlePreviousPhrase = () => {
+    const item = this.props.item;
+    if (!item || !item._value || item._value.length === 0) return;
+    const currentIdx = typeof item.playingId === "number" && item.playingId >= 0 ? item.playingId : 0;
+    const prevIdx = Math.max(currentIdx - 1, 0);
+    if (prevIdx !== currentIdx) {
+      if (item.seekToPhrase) item.seekToPhrase(prevIdx);
+      this.selectRegionsForPhrase(prevIdx);
+    }
+  };
+
+  // Select all text in the current phrase and annotate
+  handleSelectAllAndAnnotate = () => {
+    const item = this.props.item;
+    if (!item || !item._value || item._value.length === 0) return;
+    const idx = typeof item.playingId === "number" && item.playingId >= 0 ? item.playingId : 0;
+    if (item.selectAndAnnotatePhrase) {
+      item.selectAndAnnotatePhrase(idx);
+    } else if (this.selectText) {
+      // fallback: just select the text
+      this.selectText(idx);
+    }
+  };
+
+  // Get all regions for the current phrase
+  getRegionsForPhrase = (phraseIdx) => {
+    if (!isFF(FF_NER_SELECT_ALL)) return [];
+
+    const item = this.props.item;
+    if (!item || !item.annotation) return [];
+
+    const regions = item.annotation.regionStore?.regions || item.annotation.regions || [];
+    return regions.filter((region) => {
+      const start = Number.parseInt(region.start, 10);
+      const end = Number.parseInt(region.end, 10);
+      return !isNaN(start) && !isNaN(end) && start <= phraseIdx && end >= phraseIdx;
+    });
+  };
+
+  // Helper to select a region using the proper MobX-State-Tree action
+  selectRegion = (region) => {
+    if (!isFF(FF_NER_SELECT_ALL)) return false;
+
+    const item = this.props.item;
+
+    if (item.annotation.selectArea && typeof item.annotation.selectArea === "function") {
```

Comment:
What might go wrong here? I'd suggest to remove this method and use annotation method directly, possibly with optional chaining